client/client.go

```diff
@@ -392,3 +434,52 @@ func (c *Client) Deployment(deploymentID string) error {
 	}
 	return nil
 }
+
+func (c *Client) StartWebsocket() chan *ws.ProtoMsg {
+	websocketMessages := make(chan *ws.ProtoMsg, 1)
+	go func() {
+		interval := websocketReconnectionIntervalInSeconds * time.Second
+		for {
+			err := c.OpenWebsocket()
+			if err != nil {
+				log.Errorf("[%s] %s", c.MACAddress, err)
+				time.Sleep(interval)
+				continue
+			}
+			for {
+				msg, err := c.WebsocketConnection.ReadMessage()
+				if err != nil {
+					_ = c.CloseWebsocket()
+					time.Sleep(interval)
+					break
+				}
+				websocketMessages <- msg
+			}
+			continue
+		}
+	}()
+
+	return websocketMessages
+}
+
+func (c *Client) OpenWebsocket() error {
+	if !c.Config.Websocket {
+		return nil
+	}
```

Comment:
I don't understand why we do these config checks here (and in `CloseWebsocket`) instead of one context up in `Run`. If websockets are not enabled, we don't need to call `StartWebsocket` in the first place and keep the code just sleeping between `OpenWebsocket`/`CloseWebsocket`.

Reply:
You are right, it is a left-over from a previous tentative implementation, and it doesn't make sense.
I reworked it, sorry.